src/Stratis.Bitcoin/Consensus/Validators/BlockValidator.cs

```diff
@@ -165,6 +177,30 @@ public void StartPartialValidation(ChainedHeaderBlock chainedHeaderBlock, OnPart
             this.logger.LogTrace("(-)");
         }
 
+        /// <inheritdoc />
+        public PartialValidationResult ValidateSynchronously(Block block, ChainedHeader chainedHeader)
+        {
+            this.logger.LogTrace("({0}:'{1}')", nameof(block), block.GetHash());
+
+            var chainedHeaderBlock = new ChainedHeaderBlock(block, chainedHeader);
+            var validationContext = new ValidationContext { Block = block, ChainTipToExtand = chainedHeader };
+
+            this.consensusRules.PartialValidationAsync(validationContext).GetAwaiter().GetResult();
+
+            var partialValidationResult = new PartialValidationResult
```

Comment:
this can go into a method. Method can be used here and on line 145

Reply:
I dont think line 145 is right?